### ISSUE TITLE
Show progress when updating FSRS memory states

### ISSUE THREAD

**Issue by user1823** [ankidroid/Anki-Android#16642]
After updating the FSRS parameters and clicking Save, the memory states of all cards in that preset are updated.

For this, Anki shows a progress:

<img width="400" src="https://github.com/user-attachments/assets/836c6d40-36c4-4307-9636-dfba455080c9">

But, AnkiDroid just says "Processing".

Related:
- https://github.com/ankidroid/Anki-Android/issues/16354
- https://github.com/ankidroid/Anki-Android/pull/16378

**Comment by brishtibheja** [comment]
Are you talking of `optimise all presets`? I see something like this —

![Screenshot_2024-06-24-13-10-07-37_a9eef3a2a561b80d5c76daebd0f9a14c](https://github.com/ankidroid/Anki-Android/assets/136738526/3dc3f01e-fdad-42f2-9fcc-515f904763e5)

**Comment by user1823** [comment]
> Are you talking of optimise all presets?

No, I am talking about updating the parameters (by clicking `Optimize` or manually) and then clicking `Save`.

**Comment by harshit4311** [comment]
I'd love working on this issue. Could you please assign this to me?

**Comment by david-allison** [comment]
Sounds good, thanks!

**Comment by jaivsh** [comment]
Hi @david-allison, I can replace the existing "processing" tag with a progress indicator as needed. Have created an outline for the same in which i am planning to use the *\<ProgressBar\>* ui for this. Can you please assign this task to me if someone else isn't working so that i can proceed with this, since this issue seems idle for the moment.

**Comment by david-allison** [comment]
Sure, but you should look into our existing `Processing` mechanisms: 

https://github.com/ankidroid/Anki-Android/blob/ba341fc78847832b3082bfe2654d8a85cdb922db/AnkiDroid/src/main/java/com/ichi2/anki/CoroutineHelpers.kt#L334-L351

**Comment by brishtibheja** [comment]
not stale

**Comment by david-allison** [comment]
Could the main image be re-uploaded? My PC's too fast to catch the notification in Anki Desktop

**Comment by user1823** [comment]
> Could the main image be re-uploaded?

Done.

> My PC's too fast to catch the notification in Anki Desktop

What a funny drawback of having a fast PC! 😂

**Comment by david-allison** [comment]
@user1823 thanks! 

Could you check the linked PR to confirm whether you like the message I used better.

Changing it will be trivial, submitter's choice!

**Comment by user1823** [comment]
This has regressed. AnkiDroid now again shows "Processing" after clicking `Save` in the Deck Options page, instead of showing the progress (`Updating cards: 789/9097 ...`).

2.23.0alpha4

**Comment by david-allison** [comment]
Thanks so much!!!

**Comment by david-allison** [comment]
@user1823 Had a quick look and can't reproduce. Could you provide:

* A version where this was working
* Steps to reproduce, which used to work, and now fail

**Comment by user1823** [comment]
I tried the following versions and the issue was reproduced in all of them. So, maybe the issue was never fixed in the first place?
- 2.22.3
- 2.21.1
- 2.21.0 beta 5
- 2.21alpha9
- 2.21alpha8

So, I am not sure how you got the screenshot in the linked PR. 

Repro steps:
- Long press any deck on the deck picker.
- Open the Deck Options page. 
- Go to the FSRS section and slightly change any of the parameters.
- Click `Save` at the top of the page. 
- See the dialog box that says "Processing" instead of showing the progress (`Updating cards: 789/9097 ...`).

Just to note:
This is a very low priority issue because in the next Anki update, this process is going to receive atleast a 10x speedup (possibly even upto 100x). So, most users won't even see the dialog box.

**Comment by david-allison** [comment]
Been a good first issue for a while now, let's fix this.

Logcat

```
tagSocket(163) with statsTag=0xffffffff, statsUid=-1
POST: Requested /_anki/updateDeckConfigs
Displaying progress dialog: 600ms elapsed; 
cancellable: true; 
manualCancel: false
Expecting binder but got null!
ChangeSubscriber - opExecuted called with changes: # anki.collection.OpChanges@cbe2b785
browser_table: true
card: true
deck_config: true
mtime: true
study_queues: true
Fetching appWidgetIds for provider: com.ichi2.widget.deckpicker.DeckPickerWidget
AppWidgetIds to update: 
Fetching appWidgetIds for provider: com.ichi2.widget.cardanalysis.CardAnalysisWidget
AppWidgetIds to update: 
finishWithAnimation DEFAULT
SingleFragmentActivity::onPause
DeckPicker::onStart
DeckPicker::onResume
sendAnalyticsScreenView(): DeckPicker
getOptIn() status: false
updateDeckList
Refreshing deck list
onCreateOptionsMenu()
Adding badge
Adding badge
SingleFragmentActivity::onStop
onActivityFinishing(): calling cancelLocked()
SingleFragmentActivity::onDestroy
Destroying WebView
received android.intent.action.TIME_TICK
day cutoff changed null -> 1777622400
scheduled day rollover alarm at Fri May 01 09:00:00 GMT+01:00 2026 (in 13h 3m 59.982s)
day cutoff changed null -> 1777604400
DayRolloverHandler/ could not check/update day rollover: collection not open
```
